test/sequential/crowdloan.API.claiming.test.ts

```diff
@@ -0,0 +1,336 @@
+    await setCrowdloanAllocation(crowdloanRewardsAmount);
+
+    await claimRewards(crowdloanId, testUser6);
+
+    await checkUserReward(crowdloanId, testUser6);
+  });
+
+  test("CL1 is fully setup and CL2 setup the setCrowdloanAllocation + RewardVec", async () => {
+    await setCrowdloanAllocation(crowdloanRewardsAmount);
+    await initializeReward(testUser4, crowdloanRewardsAmount);
+
+    await claimRewards(crowdloanId, testUser7);
+
+    await checkUserReward(crowdloanId, testUser7);
+  });
+
+  test("CL1 is fully setup and CL2 setup the setCrowdloanAllocation + RewardVec + compleInitialization", async () => {
+    await setCrowdloanAllocation(crowdloanRewardsAmount);
+    await initializeReward(testUser4, crowdloanRewardsAmount);
+    const leaseStartBlock = await getBlockNumber();
+    const leaseEndingBlock = (await getBlockNumber()) + 5;
+    await completeInitialization(leaseStartBlock, leaseEndingBlock);
+
+    await claimRewards(crowdloanId, testUser8);
+
+    await checkUserReward(crowdloanId, testUser8);
+  });
+
+  async function checkUserReward(crowdloanId: any, user: User) {
+    const userAccountsPayable = JSON.parse(
+      JSON.stringify(
+        await api.query.crowdloan.accountsPayable(
+          crowdloanId,
+          user.keyRingPair.address
+        )
+      )
+    );
+
+    const userClaimedReward = hexToBn(
+      await userAccountsPayable.claimedReward.toString()
+    );
+
+    expect(userClaimedReward).bnEqual(crowdloanRewardsAmount);
+  }
+});
+
+async function setCrowdloanAllocation(crowdloanAllocationAmount: BN) {
+  const setCrowdloanAllocation = await Sudo.batchAsSudoFinalized(
+    Sudo.sudo(
+      api.tx.crowdloan.setCrowdloanAllocation(crowdloanAllocationAmount)
+    )
+  );
+
+  return setCrowdloanAllocation;
+}
+
+async function initializeReward(user: User, crowdloanRewardsAmount: BN) {
+  const initializeReward = await Sudo.batchAsSudoFinalized(
+    Sudo.sudo(
+      api.tx.crowdloan.initializeRewardVec([
+        [
+          user.keyRingPair.address,
+          user.keyRingPair.address,
+          crowdloanRewardsAmount,
+        ],
+      ])
+    )
+  );
+
+  return initializeReward;
+}
+
+async function completeInitialization(
+  leaseStartBlock: number,
+  leaseEndingBlock: number
+) {
+  const completeInitialization = await Sudo.batchAsSudoFinalized(
+    Sudo.sudo(
+      api.tx.crowdloan.completeInitialization(
+        leaseStartBlock,
+        // @ts-ignore
+        leaseEndingBlock
+      )
+    )
+  );
+
+  return completeInitialization;
+}
+
+async function claimRewards(crowdloanId: any, userId: User) {
+  const claimRewards = await signTx(
+    api,
+    // @ts-ignore
+    api.tx.crowdloan.claim(crowdloanId),
+    userId.keyRingPair
+  );
+  const eventResponse = getEventResultFromMangataTx(claimRewards);
+  return eventResponse;
+}
+
+async function sudoClaimRewards(crowdloanId: any, userId: User) {
+  const claimRewards = await Sudo.batchAsSudoFinalized(
+    // @ts-ignore
```

Comment:
why ts ignore?

Reply:
the wrong set of arguments in index.d.ts